language/vm/vm_runtime/src/txn_executor.rs

```diff
@@ -289,7 +289,12 @@ where
                             // 1. Rename the AccessPath here to a new type that represents such
                             //    globally unique id for event streams.
                             // 2. Charge gas for the msg emitted.
-                            self.event_data.push(ContractEvent::new(guid, count, msg))
+                            self.event_data.push(ContractEvent::new(
+                                guid,
+                                count,
+                                StructTag::Primitive(PrimitiveTypeTag::Bool),
```

Comment:
TODO for getting the type here?

Reply:
Still working on this. I'm planning to implement this properly if the tests don't require binary data to be successfully deserialized as the corresponding type.